### PR TITLE
media-sound/teamspeak-server: bump to version 3.3.0

### PR DESCRIPTION
**Pull request by ConiKost**
Closes: https://bugs.gentoo.org/659574
Package-Manager: Portage-2.3.43, Repoman-2.3.10

### CODE REVIEW

File: media-sound/teamspeak-server/files/teamspeak.service-r1 [hunk 1]
```diff
@@ -9,8 +9,8 @@ Group=teamspeak
 WorkingDirectory=/opt/teamspeak3-server
 Environment="LD_LIBRARY_PATH=/opt/teamspeak3-server"
 Environment="TS3SERVER_LICENSE=accept"
-ExecStartPre=/bin/rm -f /dev/shm/7gbhujb54g8z9hu43jre8
-ExecStart=/opt/bin/ts3server inifile=/etc/teamspeak3-server/ts3server.ini
+ExecStartPre=rm -f /dev/shm/7gbhujb54g8z9hu43jre8
+ExecStart=ts3server inifile=/etc/teamspeak3-server/ts3server.ini
```
Comment (mgorny) [inline media-sound/teamspeak-server/files/teamspeak.service-r1]: I didn't know systemd started using PATH lookup. Does it really work for /opt/bin, though?

File: media-sound/teamspeak-server/teamspeak-server-3.3.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+	opt/teamspeak3-server/ts3server
+	opt/teamspeak3-server/tsdnsserver"
+
+pkg_setup() {
+	enewgroup teamspeak
+	enewuser teamspeak -1 -1 /opt/teamspeak3-server teamspeak
+}
+
+src_unpack() {
+	unpack ${A}
```
Comment (mgorny) [inline media-sound/teamspeak-server/teamspeak-server-3.3.0.ebuild]: `default`?

File: media-sound/teamspeak-server/teamspeak-server-3.3.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+	mv teamspeak3-server_linux_$(usex amd64 amd64 x86) teamspeak3-server_linux || die
```
Comment (mgorny) [inline media-sound/teamspeak-server/teamspeak-server-3.3.0.ebuild]: If you need to rename it anyway, why not use a name that would render redefining S unnecessary?

File: media-sound/teamspeak-server/teamspeak-server-3.3.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+}
+
+src_install() {
+	touch "${T%/}"/.ts3server_license_accepted || die
+	insinto "/opt/teamspeak3-server"
+	doins "${T}"/.ts3server_license_accepted
```
Comment (mgorny) [inline media-sound/teamspeak-server/teamspeak-server-3.3.0.ebuild]: Use `dodir` and `touch` the file inside `${D}` directly. Or you could use `newins - ... <<<''` to take empty file from stdin.

File: media-sound/teamspeak-server/teamspeak-server-3.3.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+		doexe redist/libmariadb.so.2
+
+		insinto /opt/teamspeak3-server/sql
+		doins -r sql/create_mariadb
+		doins -r sql/updates_and_fixes
+	fi
+
+	if use doc; then
+		docinto serverquery/html
+		dodoc doc/serverquery/*
```
Comment (mgorny) [inline media-sound/teamspeak-server/teamspeak-server-3.3.0.ebuild]: `dodoc -r .../.`

File: media-sound/teamspeak-server/teamspeak-server-3.3.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+		docinto ts3server
+		dodoc doc/*.txt
+
+		docinto tsdns
+		dodoc tsdns/README tsdns/USAGE
+	fi
+
+	echo "CONFIG_PROTECT=\"/etc/teamspeak3-server/ts3server.ini /etc/teamspeak3-server/ts3server_mariadb.ini\"" > "${T}"/99teamspeak3-server || die
+	doenvd "${T}"/99teamspeak3-server
```
Comment (mgorny) [inline media-sound/teamspeak-server/teamspeak-server-3.3.0.ebuild]: You could use `newenvd` with stdin input here as well, and `<<<` or heredoc to make it more readable.

Comment (ConiKost) [reply media-sound/teamspeak-server/teamspeak-server-3.3.0.ebuild]: They fail with pypy for me, so removed.
-> https://pastebin.com/8Par3Ke0

File: media-sound/teamspeak-server/files/teamspeak.service-r1 (hunk 1 above)
Comment (ConiKost) [reply media-sound/teamspeak-server/files/teamspeak.service-r1]: Ouch, I wasn't aware of this. Reverted back.

File: media-sound/teamspeak-server/teamspeak-server-3.3.0.ebuild (hunk 2 above)
Comment (ConiKost) [reply media-sound/teamspeak-server/teamspeak-server-3.3.0.ebuild]: Thanks, changed.

File: media-sound/teamspeak-server/teamspeak-server-3.3.0.ebuild (hunk 3 above)
Comment (ConiKost) [reply media-sound/teamspeak-server/teamspeak-server-3.3.0.ebuild]: Done. Good idea!

File: media-sound/teamspeak-server/teamspeak-server-3.3.0.ebuild (hunk 4 above)
Comment (ConiKost) [reply media-sound/teamspeak-server/teamspeak-server-3.3.0.ebuild]: Changed.

File: media-sound/teamspeak-server/teamspeak-server-3.3.0.ebuild (hunk 5 above)
Comment (ConiKost) [reply media-sound/teamspeak-server/teamspeak-server-3.3.0.ebuild]: Done.

File: media-sound/teamspeak-server/teamspeak-server-3.3.0.ebuild (hunk 6 above)
Comment (ConiKost) [reply media-sound/teamspeak-server/teamspeak-server-3.3.0.ebuild]: Done.

**Comment by mgorny** [comment]
I don't see any of the requested changes. Did you fail to force push?

**Comment by ConiKost** [comment]
Sorry, should be now ok.

File: media-sound/teamspeak-server/teamspeak-server-3.3.0.ebuild (hunk 6 above)
Comment (mgorny) [reply media-sound/teamspeak-server/teamspeak-server-3.3.0.ebuild]: That's not what I said. 
```
newenvd - foo <<-EOF || die
...

File: media-sound/teamspeak-server/teamspeak-server-3.3.0.ebuild [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+
+src_unpack() {
+	default
+
+	mv teamspeak3-server_linux_$(usex amd64 amd64 x86) ${P} || die
+}
+
+src_install() {
+	keepdir /etc/teamspeak3-server
+	diropts -o teamspeak -g teamspeak
```
[the remark this answers is not in the document]
Comment (ConiKost) [reply media-sound/teamspeak-server/teamspeak-server-3.3.0.ebuild]: Oh, I see. What can I do differently, when I want to have `/opt/teamspeak3-server` and `/var/log/teamspeak3-server` owned by the user?

File: media-sound/teamspeak-server/teamspeak-server-3.3.0.ebuild [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+	dodir /opt/bin
+	dosym ../teamspeak3-server/ts3server-bin /opt/bin/ts3server
+
+	exeinto /opt/teamspeak3-server
+	doexe libts3db_sqlite3.so libts3_ssh.so
+
+	docinto	serverquery
+	dodoc -r serverquerydocs/.
+
+	dosym ../../usr/share/doc/teamspeak-server-${PV}/serverquery /opt/teamspeak3-server/serverquerydocs
```
[the remark this answers is not in the document]
Comment (ConiKost) [reply media-sound/teamspeak-server/teamspeak-server-3.3.0.ebuild]: Hm, good point. Will change it.

File: media-sound/teamspeak-server/teamspeak-server-3.3.0.ebuild (hunk 7 above)
[the remark this answers is not in the document]
Comment (ConiKost) [reply media-sound/teamspeak-server/teamspeak-server-3.3.0.ebuild]: I added now a new diropts, which sets back to root after that, it that's an acceptable solution?

**Comment by ConiKost** [comment]
> newenvd - foo <<-EOF || die

Ok, changed it. Thus, I don't like that type of syntax ;-)

File: media-sound/teamspeak-server/teamspeak-server-3.3.0.ebuild (hunk 7 above)
[the remark this answers is not in the document]
Comment (mgorny) [reply media-sound/teamspeak-server/teamspeak-server-3.3.0.ebuild]: Ain't pretty. Please either reset `diropts` to null, or move it to the bottom so there's nothing else affected.

[the remark this answers is not in the document]
Comment (ConiKost) [reply media-sound/teamspeak-server/teamspeak-server-3.3.0.ebuild]: Okay, set to null. Thanks.

**Comment by mgorny** [comment]
Thanks. Adding to my merging queue; will merge when I finish testing freebsd-* changes ;-).

File: media-sound/teamspeak-server/teamspeak-server-3.3.0.ebuild [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+	newins "${FILESDIR}"/ts3server.ini-r1 ts3server.ini
+
+	dodoc CHANGELOG
+	docinto ts3server
+	dodoc doc/*.txt
+
+	newinitd "${FILESDIR}"/teamspeak.initd-r1 teamspeak3-server
+	systemd_newunit "${FILESDIR}"/teamspeak.service teamspeak3-server.service
+
+	newenvd - 99teamspeak3-server <<- EOF || die
```
Comment (mgorny) [inline media-sound/teamspeak-server/teamspeak-server-3.3.0.ebuild]: Hmm, `||die` is unnecessary here. I'll fix that while merging.

**Pull request closed without merging**